README.md

```diff
@@ -1,2 +1,149 @@
+
+- When might you use your architectural style?
+- Are there advantages/disadvantages?
+- What is the simplest way to draw it?
+
+## Why is Architecture Important?
+
+Scenario 1. You're in a rush to leave home and you can't find your keys. Your room is super messy and your keys could be in 1 of 57 places. You are late and miss the first part of your morning challenge at F&C. Doh!
+
+Scenario 2. All of your belongings are arranged according to categories. There's only one place your keys could be. Life is a breeze.
+
+##### Messy systems. On a micro or macro level, they're difficult to maintain, test, extend, scale, and reason about.
+
+Messy systems are costly and they cause frustrating delays.
+
+Clean and ordered systems mitigate complexity and make your life easier in the long run.  
+
+<blockquote>
+"The quality and longevity of a software system is determined by its architecture" -- Linda Northrop in ["The Importance of Software Architecture"](http://csse.usc.edu/GSAW/gsaw2003/s13/northrop.pdf), Carnegie Mellon University.</blockquote>
+
+Be a Monica, not a Rachel.
+
+<iframe src="https://giphy.com/embed/26FPxK7LVhEHammiY" width="480" height="270" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/friends-nick-at-nite-26FPxK7LVhEHammiY"></a></p>
+
+<!-- ## Decisions, Decisions
+
+There are some major decisions that you must make when you are planning a project. Making these decisions is a key part of iteratively developing your architecture design.
+
+You need to:
+
+- Determine the Application Type
+- Determine the Deployment Strategy
+  _"Your application may be deployed in a variety of environments, each with its own specific set of constraints such as physical separation of components across different servers, a limitation on networking protocols, firewall and router configurations, and more."_
+- Determine the Appropriate Technologies
+- Determine the Quality Attributes
+  _e.g. run-time qualities, security, performance, usability, scalability_
+- Determine the Crosscutting Concerns
+  _Where can you use centralised solutions to problems that are specific to more than one layer of your application?_
+
+-- "Key Design Considerations", from the [Microsoft Application Architecture Guide](https://msdn.microsoft.com/en-us/library/ee658124.aspx#DesignConsiderations) -->
+
+## Key Principles To Keep in Mind
+
+We're going to start designing the architecture for our own app in a bit.
+
+When you are architecting an app, try to keep these key principles in mind:
+
+- How can you separate your concerns? i.e. keep unrelated things in separate sections
+- Which aspects of your app are generic, and which are specific to your requirements?
+- How can you make it easy to test?
+- How can you make it easy to change?
+- Can you anticipate changes that are likely to occur?
+
+Take a few minutes to look at your Week 2 project and come up with answers to these questions:
```

Comment:
Maybe worth seeing if the same question could be applied to the github api workshop? 

I've not reviewed it again yet so can't say off the top of my head.